flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/delegation/ParserImpl.java

```diff
@@ -70,6 +76,12 @@ public ParserImpl(
     public List<Operation> parse(String statement) {
         CalciteParser parser = calciteParserSupplier.get();
         FlinkPlannerImpl planner = validatorSupplier.get();
+
+        // use ParseStrategyParser to parse command first
+        if (ParseStrategyParser.INSTANCE.matches(statement)) {
+            return Collections.singletonList(ParseStrategyParser.INSTANCE.convert(statement));
```

Comment:
I think `ParseStrategyParser` doesn't need to expose two methods, otherwise, it will need to pattern match twice. The `ParseStrategyParser` can expose just one method:


```java
    public Optional<Operation> parse(String statement) {
        for (ExtendedParserStrategy strategy : REGEX_STRATEGIES) {
            if (strategy.match(statement)) {
                return Optional.of(strategy.convert(statement));
            }
        }
        return Optional.empty();
    }
```